### PR TITLE
show osm links for stairs

### PR DESCRIPTION
**Pull request by nnngrach**
[issue](https://github.com/osmandapp/OsmAnd-iOS/issues/3955)

before (no osm link)

Screenshot 2024-11-05 at 02 12 09

after (with osm link)

Screenshot 2024-11-05 at 02 07 41

osm links for another poi still correct

Screenshot 2024-11-05 at 02 08 47

for ways too

Screenshot 2024-11-05 at 02 15 52

### CODE REVIEW

File: Sources/Map/ObfConstants.swift [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    static private let SHIFT_ID = 6
+    static private let AMENITY_ID_RIGHT_SHIFT = 1
+    static private let WAY_MODULO_REMAINDER = 1
+    
+    static private let NODE = "node"
+    static private let WAY = "way"
+    static private let RELATION = "relation"
+    
+    static func getOsmUrlForId(_ object: OAPOI) -> String {
+        if let type = getOsmEntityType(object) {
```
Comment (tigrim) [inline Sources/Map/ObfConstants.swift:34]: guard

File: Sources/Map/ObfConstants.swift [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            var osmId = getOsmObjectId(object)
+            return "https://www.openstreetmap.org/" + type + "/" + String(osmId)
```
Comment (tigrim) [inline Sources/Map/ObfConstants.swift:36]: return "https://www.openstreetmap.org/\(type)/\(osmId)"

File: Sources/Map/ObfConstants.swift [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                return WAY
+            }
+            if isIdFromPropagatedNode(obfId) {
+                return WAY
+            }
+            var relationShift = 1 << 41
+            if originalId > relationShift {
+                return RELATION
+            } else {
+                return obfId % 2 == Self.WAY_MODULO_REMAINDER ? WAY : NODE
```
Comment (tigrim) [inline Sources/Map/ObfConstants.swift:77]: https://developer.apple.com/documentation/swift/int/ismultiple(of:)

File: Sources/Controllers/TargetMenu/POI/OAPOIViewController.mm [hunk 4]
```diff
@@ -795,18 +795,19 @@ - (void) buildRows:(NSMutableArray<OARowInfo *> *)rows
...
 
     long long objectId = self.poi.obfId;
-    if (osmEditingEnabled && (objectId > 0 && ((objectId % 2 == AMENITY_ID_RIGHT_SHIFT) || (objectId >> NON_AMENITY_ID_RIGHT_SHIFT) < INT_MAX)))
+    if (osmEditingEnabled && (objectId > 0 && ((objectId % 2 == AMENITY_ID_RIGHT_SHIFT) || (objectId >> NON_AMENITY_ID_RIGHT_SHIFT) < INT_MAX || self.poi.isRenderedObject)))
     {
         OAPOIType *poiType = self.poi.type;
         BOOL isAmenity = poiType && ![poiType isKindOfClass:[OAPOILocationType class]];
 
         long long entityId = objectId >> (isAmenity ? AMENITY_ID_RIGHT_SHIFT : NON_AMENITY_ID_RIGHT_SHIFT);
         BOOL isWay = objectId % 2 == WAY_MODULO_REMAINDER; // check if mapObject is a way
```
Comment (tigrim) [inline Sources/Controllers/TargetMenu/POI/OAPOIViewController.mm:804]: isMultiple

File: Sources/Map/ObfConstants.swift [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+//
+//  Created by Max Kojin on 04/11/24.
+//  Copyright © 2024 OsmAnd. All rights reserved.
+//
+
+import Foundation
+
+@objc(OAObfConstants)
+@objcMembers
+class ObfConstants: NSObject {
```
Comment (tigrim) [inline Sources/Map/ObfConstants.swift:13]: final

File: Sources/Controllers/Map/Layers/OAPOILayer.mm [hunk 6]
```diff
@@ -479,6 +479,7 @@ - (void) collectObjectsFromPoint:(CLLocationCoordinate2D)point touchPoint:(CGPoi
             }
             else
             {
+                poi.isRenderedObject = YES;
```
Comment (alex-osm) [inline Sources/Controllers/Map/Layers/OAPOILayer.mm:482]: Do not looks right. Why `obfsDataInterface->findAmenityByObfMapObject `did not find amenity?

**Pull request merged**